openedx/core/djangoapps/credit/services.py

```diff
@@ -0,0 +1,130 @@
+"""
+Implementation of "credit" XBlock service
+"""
+
+import logging
+
+from django.contrib.auth.models import User
+from django.core.exceptions import ObjectDoesNotExist
+
+from opaque_keys.edx.keys import CourseKey
+
+from student.models import CourseEnrollment, UserProfile
+
+log = logging.getLogger(__name__)
+
+
+class CreditService(object):
+    """
+    Course Credit XBlock service
+    """
+
+    def get_credit_state(self, user_id, course_key_or_id):
+        """
+        Return all information about the user's credit state inside of a given
+        course.
+
+        ARGS:
+            - user_id: The PK of the User in question
+            - course_key: The course ID (as string)
+
+        RETURNS:
+            NONE (user not found or is not enrolled or is not credit course)
+            - or -
+            {
+                'enrollment_mode': the mode that the user is enrolled in the course
+                'profile_fullname': the name that the student registered under, used for verification
+                'credit_requirement_status': the user's status in fulfilling those requirements
+            }
+        """
+
+        # This seems to need to be here otherwise we get
+        # circular references when starting up the app
+        from openedx.core.djangoapps.credit.api.eligibility import (
+            is_credit_course,
+            get_credit_requirement_status,
+        )
+
+        # course_key = CourseKey.from_string(course_id)
```

Comment:
Is this line necessary to keep here for some reason?  If so please include a comment describing the use case in whihc to enable the line.  If not, please remove the line.